smartsnippets/views.py

```diff
@@ -1 +1,95 @@
-# Create your views here.
+import json
+
+from django.contrib.admin.templatetags.admin_static import static
+from django.contrib.sites.models import Site
+from django.core.urlresolvers import reverse
+from django.forms.widgets import Media as WidgetsMedia
+from django.shortcuts import get_object_or_404, render_to_response
+from django.template import RequestContext
+
+from cms.models import Page
+
+from smartsnippets.cms_plugins import variables_media
+from smartsnippets.models import SmartSnippet, SmartSnippetPointer, Variable
+
+
+def get_snippet_edit_code(request, snippet_id, config=None):
+    """
+    Returns a HTTPResponse that renders the admin of a smartsnippet.
+
+    :param request: Request needed to create the rendering context
+    :param snippet_id: id of the smartsnippet model
+    :param config: dictionary containing existing configuration of the smartsnippet instance,
+                   None if new
+    Schema: {
+      'metadata': {
+        'snippet_id': snippet_id,
+      },
+      'variables':{
+        smart_snipet_variable_name: deserialized_variable_data,
+      },
+    }
+    """
+    snippet = get_object_or_404(SmartSnippet, id=snippet_id)
+    snippet_vars = snippet.variables.all()
+    fake_pointer = SmartSnippetPointer(snippet=snippet)
+    current_site = Site.objects.get_current()
+    page = Page(site=current_site)
+    var_order = lambda var: (var._order, var.name)
+    variables_data = config.get('variables', {}) if config else {}
+
+    def make_var(snippet_var):
+        var = Variable(snippet=fake_pointer, snippet_variable=snippet_var)
+        if snippet_var.name in variables_data:
+            var.value = json.dumps(variables_data[snippet_var.name])
+        return var
+
+    variables = [
+        make_var(snippet_var) for snippet_var in sorted(snippet_vars, key=var_order)
```

Comment:
nitpicking, can you do this in line 34 with .order_by with respect to ('_order', 'name') ?